### ISSUE TITLE
[FIRRTL] Fix the syntax of `firrtl.constant` to match `rtl.constant`

### ISSUE THREAD

**Issue by lattner** [llvm/circt#762]
We currently use:

```
 %c5_ui8 = firrtl.constant(5 : ui8) : !firrtl.uint<8>
```
it would be waaay nicer to have:

```
 %c5_ui8 = firrtl.constant 5 : !firrtl.uint<8>
```

Also, the implementation of this node sometimes stores the attributes with i8 type, sometimes with ui8, etc.  It would be good to make this consistent and verify it.

**Comment by Ayushsunny** [comment]
Hey @lattner I wanna fix this

**Comment by lattner** [comment]
Sure, please create a PR!

**Comment by Ayushsunny** [comment]
> Sure, please create a PR!
Sir, I have made a PR for this issue 
could you please review it https://github.com/llvm/circt/pull/954

**Comment by lattner** [comment]
For others following along, the direction proposed isn't the right patch. I don't consider this bug to be taken by anyone.

**Comment by lattner** [comment]
This is actually surprisingly difficult because of the way `parseAttribute` works, but [this MLIR patch](https://reviews.llvm.org/D102120) will allow addressing this.

**Comment by lattner** [comment]
Finally implemented in 2d1c29e8
